demos/yielding/README.md

```diff
@@ -0,0 +1,47 @@
+# Yielding
+
+In this demo we show the usage of yielding, which occurs in a computer program during multithreading.
+
+By calling syscall `yield`, the thread gives away the utilization of the CPU to another thread, or a specific thread (discussed in another demo).
+
+To describe thread yielding, we need to introduce:
+
+* [`add_thread_from_worker`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/user/thread.c#L42)
+* [`do_scheduler`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/kernel/thread.c#L353)
+* [`yield`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/user/thread.c#L92)
+
+## [`add_thread_from_worker`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/user/thread.c#L42)
+
+In AMT we store all of the threads in an array called `all_threads`.
+
+When a thread calls this function, the kernel will create a new thread with the given arguments, initialize the permissions, and set its starting point to be the given function (would be `thread_worker()` in this demo).
+
+After the creation of the thread, it will wait to be chosen by the scheduler so it can utilize the CPU.
+
+## [`do_scheduler`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/kernel/thread.c#L353)
+
+The job of the scheduler is to choose the next thread to make use of the CPU. 
+
+In AMT, the scheduler simply walks through the `all_threads` array to find the next thread.
+
+Then the task of the thread to be switched will be suspended until the thread has been chosen to run.
```

Comment:
"Then the the current thread will be suspended until it is chosen to run again."

- "the current thread" is clearer than "task to be switched" for me. The latter could mean the current or the next thread.
- Remove "task of" since if the thread is suspended, then it's task/work code is implicitly also suspended
- add "again" on the end, to emphasize that it will be resuming not running for the first time.

Side note: When we make a syscall the "current thread" is technically suspended but I still call it the current thread until `current_thread` actually changes.